mono/metadata/profiler.c

```diff
@@ -772,3 +785,78 @@ mono_profiler_set_events (int flags)
 {
 	/* Do nothing. */
 }
+
+static void
+allocation_cb (MonoProfiler *prof, MonoObject* object)
+{
+	prof->allocation (prof->profiler, object, object->vtable->klass);
+}
+
+void
+mono_profiler_install_allocation (MonoLegacyProfileAllocFunc callback)
+{
+	current->allocation = callback;
+
+	if (callback)
+		mono_profiler_set_gc_allocation_callback (current->handle, allocation_cb);
+}
+
+static void
+enter_cb (MonoProfiler *prof, MonoMethod *method, MonoProfilerCallContext *context)
+{
+	prof->enter (prof->profiler, method);
+}
+
+static void
+leave_cb (MonoProfiler *prof, MonoMethod *method, MonoProfilerCallContext *context)
+{
+	prof->leave (prof->profiler, method);
+}
+
+void
+mono_profiler_install_enter_leave (MonoLegacyProfileMethodFunc enter, MonoLegacyProfileMethodFunc fleave)
+{
+	current->enter = enter;
+	current->leave = fleave;
+
+	if (enter)
+		mono_profiler_set_method_enter_callback (current->handle, enter_cb);
+
+	if (fleave)
+		mono_profiler_set_method_leave_callback (current->handle, leave_cb);
+}
+
+static void
+throw_callback_cb (MonoProfiler *prof, MonoObject *exception)
+{
+	prof->throw_callback (prof->profiler, exception);
+}
+
+static void
+exc_method_leave_cb (MonoProfiler *prof, MonoMethod *method, MonoObject *exception)
+{
+	prof->leave (prof->profiler, method);
+}
+
+static void
+clause_callback_cb (MonoProfiler *prof, MonoMethod *method, uint32_t index, MonoExceptionEnum type, MonoObject *exception)
+{
+	prof->clause_callback (prof->profiler, method, type, index);
+}
+
+void
+mono_profiler_install_exception (MonoLegacyProfileExceptionFunc throw_callback, MonoLegacyProfileMethodFunc exc_method_leave, MonoLegacyProfileExceptionClauseFunc clause_callback)
+{
+	current->throw_callback = throw_callback;
+	current->exc_method_leave = exc_method_leave;
+	current->clause_callback = clause_callback;
+
+	if (throw_callback)
+		mono_profiler_set_exception_throw_callback (current->handle, throw_callback_cb);
+
+	if (exc_method_leave)
+		mono_profiler_set_method_exception_leave_callback (current->handle, exc_method_leave_cb);
+
+	if (clause_callback)
+		mono_profiler_set_exception_clause_callback (current->handle, clause_callback_cb);
+}
```

Comment:
(Add a newline to avoid unnecessary diffs in the future from editors that do this automatically.)